optimism/src/mips/interpreter.rs

```diff
@@ -243,7 +249,17 @@ pub fn interpret_itype<Env: InterpreterEnv>(env: &mut Env, instr: ITypeInstructi
         ITypeInstruction::LoadUpperImmediate => (),
         ITypeInstruction::Load8 => (),
         ITypeInstruction::Load16 => (),
-        ITypeInstruction::Load32 => (),
+        ITypeInstruction::Load32 => {
+            let rt = env.get_instruction_part(InstructionPart::RT);
+            // Values are the first 16bits
+            let immediate_value = (env.get_instruction_part(InstructionPart::RD) << 11)
+                + (env.get_instruction_part(InstructionPart::Shamt) << 6)
+                + (env.get_instruction_part(InstructionPart::Funct));
+            env.overwrite_register_checked(&rt, &immediate_value);
+            env.set_instruction_pointer(env.get_instruction_pointer() + Env::constant(4u32));
```

Comment:
So the reason that you increase by 4 the instruction pointer is that instructions take 32 bits in memory (i.e. 4 bytes) and when you read one instruction you move to the next?

Reply:
y